R/sample_predictions.R

```diff
@@ -39,47 +43,62 @@
 #'   point_nowcast_matrix,
 #'   reporting_triangle,
 #'   disp,
-#'   fun_to_aggregate = sum,
-#'   k = 2
+#'   ref_time_aggregator = function(x) zoo::rollsum(x, k = 2, align = "right")
 #' )
 #' nowcast_pred_draw_agg
-sample_prediction <- function(point_nowcast_matrix,
-                              reporting_triangle,
-                              dispersion,
-                              fun_to_aggregate = sum,
-                              k = 1) {
-  .validate_aggregation_function(fun_to_aggregate)
-  if (length(dispersion) > nrow(point_nowcast_matrix)) {
+sample_prediction <- function(
+    point_nowcast_matrix,
+    reporting_triangle,
+    uncertainty_params,
+    error_model = function(pred, params) sample_nb(pred, params),
```

Comment:
pass sample_nb directly wrapper doesn't do anything

Reply:
Specifically, what I tried was : `error_model = sample_nb()` where the function args all had defaults (sometimes NULL)

Reply:
error_model = sample_nb() this is calling not passing the function